calculator/src/main/java/com/market/banica/calculator/service/CalculatorServiceImpl.java

```diff
@@ -28,18 +35,22 @@
     @Override
     public RecipeDTO getRecipe(String clientId, String itemName, int quantity) {
 
-        throw new FeatureNotSupportedException("Feature is not implemented yet.");
+        List<Product> products = productService.getProductAsListProduct(itemName);
+        List<ItemOrderBookResponse> resultList = new ArrayList<>();
 
-        //Due the connection is fake atm.
-        //method will fail because of the lack of real connection to aurora service.
+        for (int i = 0; i < products.size(); i++) {
 
-        //get recipe from property
+            resultList.add(auroraService.getIngredient(itemName, clientId));
 
-        //get desired ingredients name
+        }
 
-        //call aurora service for specific ingredient price.
+        ItemOrderBookResponse product = resultList.get(0);
 
-        //IngredientResponse ingredients = auroraService.getIngredient(ingredient, quantity);
+        RecipeDTO result = new RecipeDTO();
+        result.setItemName(product.getItemName());
+        result.setIngredients(null);
+        result.setTotalPrice(BigDecimal.valueOf(product.getOrderbookLayersList().get(0).getPrice()));
```

Comment:
null check on product.getOrderbookLayersList and on product.getOrderbookLayersList().get(0)

Reply:
The calculator service class code is added for the sole purpose of integration tests, it connects the rest services with data from the order book service, it is now being reworked in the new branch for food item price optimization logic. It has no intention to represent real use case scenario but to mock the behaviour of the real class to be implemented.